helix-rest/src/main/java/org/apache/helix/rest/server/service/InstanceServiceImpl.java

```diff
@@ -206,14 +216,27 @@ public StoppableCheck getInstanceStoppableCheck(String clusterId, String instanc
           instancesForCustomPartitionLevelChecks, restConfig, customPayLoads);
       for (Map.Entry<String, StoppableCheck> instancePartitionStoppableCheckEntry : instancePartitionLevelChecks
           .entrySet()) {
-        finalStoppableChecks.put(instancePartitionStoppableCheckEntry.getKey(),
-            instancePartitionStoppableCheckEntry.getValue());
+        String instance = instancePartitionStoppableCheckEntry.getKey();
+        StoppableCheck stoppableCheck = instancePartitionStoppableCheckEntry.getValue();
+        addStoppableCheck(finalStoppableChecks, instance, stoppableCheck);
       }
     }
 
     return finalStoppableChecks;
   }
 
+  private void addStoppableCheck(Map<String, StoppableCheck> stoppableChecks, String instance,
+      StoppableCheck stoppableCheck) {
+    if (!stoppableChecks.containsKey(instance)) {
+      stoppableChecks.put(instance, stoppableCheck);
+    } else {
+      // Merge two checks
+      StoppableCheck existedCheck = stoppableChecks.get(instance);
+      existedCheck.add(stoppableCheck);
+      stoppableChecks.put(instance, existedCheck);
+    }
```

Comment:
NIT: Can be replaced with `computeIfAbsent`.

Reply:
I thought about both `putIfAbsent` and `computeIfAbsent` but neither of them fit in the logic.

Reply:
Sorry, not `computeIfAbsent`, but `compute`. Do you think this would work?
`stoppableChecks.compute(instance, (k, v) -> v == null ? stoppableCheck : v.add(stoppableCheck)`

Reply:
I thought about it. But `v.add(stoppableCheck)` would require the `add()` method to return `this` object which I don't prefer to.